src/background.js

```diff
@@ -18,6 +19,14 @@ import store from "./media-manager/store";
 
 const isDevelopment = process.env.NODE_ENV !== "production";
 
+// We need to ask macOS permission to access media devices
+async function getMediaPermission() {
+  await systemPreferences.askForMediaAccess("microphone");
```

Comment:
We should only do this for MacOS and also check the result to provide a message to the end-user (as you suggested) and not just take this for granted. 

Taking the code from https://lifesaver.codes/answer/systempreferences-askformediaaccess(mediatype)-causes-app-to-crash-in-macos-mojave

```javascript
async function askForMediaAccess(): Promise<boolean> {
  try {
    if (platform !== "darwin") {
      return true;
    }

    const status = await systemPreferences.getMediaAccessStatus("microphone");
    log.info("Current microphone access status:", status);

    if (status === "not-determined") {
      const success = await systemPreferences.askForMediaAccess("microphone");
      log.info("Result of microphone access:", success.valueOf() ? "granted" : "denied");
      return success.valueOf();
    }

    return status === "granted";
  } catch (error) {
    log.error("Could not get microphone permission:", error.message);
  }
  return false;
}
```

Reply:
Thanks @TimPietrusky.

The only other thing I would say is that we might need to check Windows 10 media access status too as it's possible media access was restricted at a global system level according to the Electron docs:

https://www.electronjs.org/docs/api/system-preferences#systempreferencesgetmediaaccessstatusmediatype-windows-macos

But we can split the functionality per platform quite easily.